### ISSUE TITLE
java: fix temurin install symlinks

### ISSUE THREAD

**Issue by vadimvera** [jdx/mise#1494]
I'm not entirely sure when this changed because `mise` is evolving quickly, but I clearly remember when I used `rtx` that all the `Java` installs had this target links for their corresponding `major` and `major.minor` versions. For instance, `Node.js` does it (and many others):

```shell
[vadim@arcane-hub:~]$ ll .local/share/mise/installs/node/
total 12K
lrwxrwxrwx 1 vadim    9 Jan 15 18:41 16 -> ./16.20.2/
lrwxrwxrwx 1 vadim    9 Jan 15 18:41 16.20 -> ./16.20.2/
drwxr-xr-x 6 vadim 4.0K Jan 15 18:39 16.20.2/
lrwxrwxrwx 1 vadim    9 Jan 15 18:41 18 -> ./18.19.0/
lrwxrwxrwx 1 vadim    9 Jan 15 18:41 18.19 -> ./18.19.0/
drwxr-xr-x 6 vadim 4.0K Jan 15 18:39 18.19.0/
lrwxrwxrwx 1 vadim    9 Jan 15 18:41 20 -> ./20.11.0/
lrwxrwxrwx 1 vadim    9 Jan 15 18:41 20.11 -> ./20.11.0/
drwxr-xr-x 6 vadim 4.0K Jan 15 18:39 20.11.0/
lrwxrwxrwx 1 vadim    9 Jan 15 18:41 latest -> ./20.11.0/
lrwxrwxrwx 1 vadim    9 Jan 15 18:41 lts -> ./20.11.0/
lrwxrwxrwx 1 vadim    9 Jan 15 18:41 lts-gallium -> ./16.20.2/
lrwxrwxrwx 1 vadim    9 Jan 15 18:41 lts-hydrogen -> ./18.19.0/
lrwxrwxrwx 1 vadim    9 Jan 15 18:41 lts-iron -> ./20.11.0/

# ...but Java does not
[vadim@arcane-hub:~]$ ll .local/share/mise/installs/java/
total 16K
drwxr-xr-x 8 vadim 4.0K Jan 21 09:29 temurin-8.0.402+6/
drwxr-xr-x 9 vadim 4.0K Jan 21 09:30 temurin-11.0.22+7/
drwxr-xr-x 9 vadim 4.0K Jan 21 09:30 temurin-17.0.10+7/
drwxr-xr-x 9 vadim 4.0K Jan 21 09:30 temurin-21.0.2+13.0.LTS/
```

Would be great to have those back because they are quite convenient to use in many places where a given major version can be used.

`mise doctor` output:

```shell
[vadim@arcane-hub:~]$ mise doctor 
mise version:
  2024.1.11 linux-x64 (2024-01-15)

build:
  Target: x86_64-unknown-linux-gnu
  Features: DEFAULT, NATIVE_TLS
  Built: Mon, 15 Jan 2024 14:19:45 +0000
  Rust Version: rustc 1.75.0 (82e1608df 2023-12-21) (built from a source tarball)
  Profile: release

shell:
  /bin/bash
  GNU bash, version 5.2.21(1)-release (x86_64-pc-linux-gnu)
  Copyright (C) 2022 Free Software Foundation, Inc.
  License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

  This is free software; you are free to change and redistribute it.
  There is NO WARRANTY, to the extent permitted by law.

mise data directory:
  /home/vadim/.local/share/mise

mise environment variables:
  MISE_USE_TOML=1
  MISE_SHELL=bash

mise settings file:
  ~/.config/mise/settings.toml

settings:
  {
    "all_compile": false,
    "always_keep_download": false,
    "always_keep_install": false,
    "asdf_compat": false,
    "color": true,
    "disable_default_shorthands": false,
    "disable_tools": [],
    "experimental": false,
    "jobs": 4,
    "legacy_version_file": true,
    "legacy_version_file_disable_tools": [],
    "node_compile": false,
    "not_found_auto_install": true,
    "paranoid": false,
    "plugin_autoupdate_last_check_duration": "7d",
    "python_compile": false,
    "python_venv_auto_create": false,
    "raw": false,
    "shorthands_file": null,
    "task_output": null,
    "trusted_config_paths": [],
    "quiet": false,
    "verbose": false,
    "yes": false,
    "ci": false,
    "cd": null,
    "debug": false,
    "env_file": null,
    "trace": false,
    "log_level": "info"
  }

config files:
  /home/vadim/.config/mise/config.toml

plugins:
  bun             (core)
  deno            (core)
  go              (core)
  java            (core)
  node            (core)
  python          (core)
  ruby            (core)

toolset:
  go@latest, bun@latest, deno@latest, java@temurin-21, node@20

No problems found
```

**Comment by jdx** [comment]
I'm reasonably certain this has never worked for temurin due to the syntax of their versions. It works for other java versions though.

I'm going to alter the title to fit the issue.

**Comment by roele** [comment]
@jdx This seems to work now, possibly fixed by [8d39995](https://github.com/jdx/mise/commit/8d39995e615527ba7187b3d25369a506bcb21e0c)
